### PR TITLE
Test non-rootfs partition enc-decryption

### PR DESCRIPTION
**Pull request by DaniilKl**
The test tests encryption and decryption using TPM of logical partition.

### CODE REVIEW

File: dasharo-security/tpm2-commands.robot [hunk 1]
```diff
@@ -225,6 +225,44 @@ TPMCMD011.001 Performing HMAC operation on the file (Ubuntu 22.04)
...
+    ...    TPM.
+    # Check:
+    ${out}=    Execute Linux Command    apt info tpm2-tools
+    ${passed}=    Run Keyword And Return Status    "Installed: Yes"
+    ...    Should Contain    ${out}    TPM2_EncryptDecrypt
+    ${out}=    Execute Linux Command    apt info cryptsetup
+    ${passed}=    Run Keyword And Return Status
+    ...    Should Contain    ${out}    "Installed: yes"
+    # Prepare:
+    Execute Linux Command    fallocate -l 20MB test-partition
```
Comment (macpijan) [inline dasharo-security/tpm2-commands.robot:239]: @DaniilKl Please read: https://github.com/Dasharo/open-source-firmware-validation#guidelines

Comment (DaniilKl) [reply dasharo-security/tpm2-commands.robot]: @macpijan My bad, thanks for pointing out. I have added a commit witch fixes the test so as to match the guidelines: 74e157f93ce40235c75738ceccff1e194c756448. 

I have some CI's failed on my latest commit. Two of the failed CI's report problems with `Secure-Boot` and the other two with `custom boot menu key`. Both of the errors are out of the scope of my changes.

File: dasharo-security/tpm2-commands.robot [hunk 2]
```diff
@@ -225,6 +225,39 @@ TPMCMD011.001 Performing HMAC operation on the file (Ubuntu 22.04)
...
+    Execute Command In Terminal    cryptsetup luksOpen --key-file=key ./test-partition test-partition
+    # Test by adding a file:
+    Execute Command In Terminal    mkfs.ext4 /dev/mapper/test-partition
+    Execute Command In Terminal    mount /dev/mapper/test-partiton /mnt
+    Execute Command In Terminal    touch /mnt/hello-world
+    Execute Command In Terminal    umount /mnt
+    Execute Command In Terminal    cryptsetup luksClose test-partition
+    Execute Command In Terminal    cryptsetup luksOpen ./test-partition --key-file=key test-partition
+    Execute Command In Terminal    mount /dev/mapper/test-partition /mnt
+    Execute Command In Terminal    ls /mnt | grep hello-world
```
Comment (m-iwanicki) [inline dasharo-security/tpm2-commands.robot:253]: @DaniilKl   Execute Command In Terminal doesn't check if command finished successfully, you should probably save returned output and check that it contains `hello-world`

**Review by m-iwanicki — CHANGES_REQUESTED** [review]

File: dasharo-security/tpm2-commands.robot [hunk 3]
```diff
@@ -225,6 +225,39 @@ TPMCMD011.001 Performing HMAC operation on the file (Ubuntu 22.04)
     Should Contain    ${out1}    hmac.out
     Should Not Contain    ${out2}    hmac.out
 
+TPMCMD012.001 Encrypt and Decrypt non-rootfs partition (Ubuntu 22.04)
+    [Documentation]    Test encrypting and decrypting non-rootfs partition using
+    ...    TPM.
+    # Check:
+    Check If Package Is Installed    cryptsetup
```
Comment (m-iwanicki) [inline dasharo-security/tpm2-commands.robot:232]: @DaniilKl This keyword returns TRUE or FALSE depending on if package is installed. You should fail if returned value is FALSE.

**Review by m-iwanicki — CHANGES_REQUESTED** [review]

**Review by m-iwanicki — CHANGES_REQUESTED** [review]
Ignore this problem it doesn't happen because of this PR.

~~There is also problem with robotidy that happened to you and me when I tried to cherry pick your commits. 
It also happens every time now when I try to add new commits. It should probably be resolved before merging because it would mean every new commit would fail pre-commit checks~~

<details><summary>Error Log</summary>
    <pre>
robotidy.................................................................Failed
- hook id: robotidy
- exit code: 1

Traceback (most recent call last):
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/bin/robotidy", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/rich_click/rich_command.py", line 126, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/decorators.py", line 30, in wrapper
    raise err
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/decorators.py", line 21, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/cli.py", line 480, in cli
    global_config = config_module.MainConfig(cli_config)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/config.py", line 228, in __init__
    self.default_loaded = Config.from_raw_config(self.default)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/config.py", line 381, in from_raw_config
    return cls(
           ^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/config.py", line 323, in __init__
    self.load_transformers(transformers_config, force_order, target_version, skip)
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/config.py", line 402, in load_transformers
    transformers = load_transformers(
                   ^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/transformers/__init__.py", line 480, in load_transformers
    for container in import_transformer(name, transformers_config, skip):
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/transformers/__init__.py", line 324, in import_transformer
    yield create_transformer_instance(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/transformers/__init__.py", line 346, in create_transformer_instance
    positional, named, argument_names = resolve_args(short_name, spec, args, skip, handles_skip=handles_skip)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/transformers/__init__.py", line 424, in resolve_args
    argument_names = resolve_argument_names(spec.argument_names, handles_skip)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miwanicki/.cache/pre-commit/repodhj41rs4/py_env-python3/lib/python3.11/site-packages/robotidy/transformers/__init__.py", line 368, in resolve_argument_names
    return new_args + argument_names
           ~~~~~~~~~^~~~~~~~~~~~~~~~
TypeError: can only concatenate list (not "tuple") to list
Fatal exception occurred. You can create an issue at https://github.com/MarketSquare/robotframework-tidy/issues . Thanks!
    </pre>
</details>

File: dasharo-security/tpm2-commands.robot (hunk 2 above)
Comment (DaniilKl) [reply dasharo-security/tpm2-commands.robot:253]: @TomaszAIR Note that Ansible support have not been merget yet, so we do not have Ansible support on this brunch.

**Comment by DaniilKl** [comment]
Rebased on the new base: `encrypted-rootfs-release`.

File: dasharo-security/tpm2-commands.robot (hunk 3 above)
Comment (DaniilKl) [reply dasharo-security/tpm2-commands.robot:292]: @m-iwanicki This should be checked in Ansible. Waiting for Ansible support to be released.

File: dasharo-security/tpm2-commands.robot (hunk 2 above)
Comment (TomaszAIR) [reply dasharo-security/tpm2-commands.robot:313]: @DaniilKl ansible support is added now into `encrypted-rootfs-release` branch. You can use it by rebasing this work.

Comment (DaniilKl) [reply dasharo-security/tpm2-commands.robot]: @m-iwanicki From now on it is checked by KW `Should Contain`, amended here e99207d1747d267dcdee7e33aff10d5bb1955fd7.

@TomaszAIR Added ansible support for this test here da7c1090fab45e49b61f0f47551256f579cffc19.

**Comment by TomaszAIR** [comment]
I rebased once again and fixed ansible to match latest changes in this regard.

Test result

```
 λ robot -b file.txt -L TRACE -v ansible_config:yes -v rte_ip:127.0.0.1 -v snipeit:no -v config:qemu dasharo-security/tpm2-commands.robot
==============================================================================
Tpm2-Commands                                                                 
==============================================================================
                                                                                                                       
Checking if tpm2-tools is installed...                                                                                 
                                                                                                                       
Package tpm2-tools is installed
TPMCMD001.001 Check if both SHA1 and SHA256 PCRs are enabled (Ubun... | PASS |
------------------------------------------------------------------------------
TPMCMD002.001 PCRREAD Function Verification (Ubuntu 22.04) :: This... | PASS |
(...)
------------------------------------------------------------------------------
TPMCMD014.001 Encrypt and Decrypt non-rootfs partition (Ubuntu 22.... | PASS |
------------------------------------------------------------------------------
Tpm2-Commands                                                         | PASS |
18 tests, 17 passed, 0 failed, 1 skipped
==============================================================================
```

Merging.

**Pull request merged**